### PR TITLE
Configure Renovate

### PR DESCRIPTION
**Pull request by tklauser**
Renovate is a bot that allows to update dependencies in the repository.

The configuration is based on the one from the cilium/ciliume repo: https://github.com/cilium/cilium/blob/030bb1abbdb0a3bf6de7eae2f53af6ea6939c1d9/.github/renovate.json5

For future reference the documentation is available in:
- https://github.com/renovatebot/tutorial
- https://docs.renovatebot.com/configuration-options

### CODE REVIEW

File: .github/workflows/multicluster.yaml [hunk 1]
```diff
@@ -83,6 +83,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9
         with:
+          # renovate: datasource=golang-version depName=go
           go-version: 1.20.2
```
Comment (bimmlerd) [inline .github/workflows/multicluster.yaml:87]: complete drive-by but I learned recently that we should probably be quoting these versions: https://xeiaso.net/blog/go-1.20-yaml

File: .github/renovate.json5 [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        "docker.io/library/golang"
+      ],
+      "schedule": [
+        "on friday"
+      ]
+    },
+    {
+      // Group golangci-lint updates to overrule grouping of version updates in the GHA files.
+      // Without this, golangci-lint updates are not in sync for GHA files and other usages.
+      "groupName": "golangci-lint",
```
Comment (bimmlerd) [inline .github/renovate.json5:105]: do we also need to annotate the `GOLANGCILINT_WANT_VERSION` in the Makefile, or is that autodetected? https://github.com/cilium/cilium-cli/blob/main/Makefile#L16

Similarly, I guess the Go image in the Makefile is autodetected?

**Review by bimmlerd — APPROVED** [review]
LGTM, question and drive-by nit

File: .github/workflows/multicluster.yaml (hunk 1 above)
Comment (tklauser) [reply .github/workflows/multicluster.yaml:87]: Good point. I guess we're not affected by this though because we always specify complete 1.x.y versions, not 1.x versions which might be (mis)interpreted as float. I'll leave the version as is for the sake of this PR and will send a follow up fix.

File: .github/renovate.json5 (hunk 2 above)
Comment (tklauser) [reply .github/renovate.json5:105]: Good point, added.

**Review by brlbil — APPROVED** [review]

**Review by michi-covalent — APPROVED** [review]
🤖
